dbmanager/src/dbmanager.c

```diff
@@ -21,18 +22,131 @@ struct DB {
   dbNode* head;
 };
 
+
+/*  function to create a new entry for the database.
+ *  @key - a unique integer value.
+ *  @val - pointer to the data stored.
+ *  @return - pointer if success, NULL if failure.
+ */
+static dbNode* create_node(int key, void* val) {
+  dbNode* newNode = (dbNode*)malloc(sizeof(dbNode));
+  return newNode;
+}
+
+
 /*
- *  keeps track of head of database.
+ *  deletes all the nodes from the given tree.
+ *  @head - head of the tree to be deleted.
  */
-dbNode* dbHead = NULL;
+static void delete_tree(dbNode* head) {
 
-/*  function to create a new entry for the database.
+}
+
+
+/*
+ *  inserts node into the given tree.
+ *  @dbCxt - database context to operate on.
+ *  @node - node to be inserted.
+ */
+static void insert_node(DB* dbCxt, dbNode* node) {
+   
+}
+
+
+/*
+ *  deletes node from the given tree.
+ *  @dbCxt - database context to operate on.
+ *  @key - key to be deleted.
+ *  @return - 0 on success, -1 on failure to delete.
+ */
+static int delete_node(DB* dbCxt, int key) {
+  return 0;
+}
+
+
+/*
+ *  returns value corresponding to provided key from the given tree.
+ *  @dbCxt - database context to operate on.
+ *  @key - key to be searched for.
+ *  @return - value pointer on success, NULL on failure to find key.
+ */
+static void* get_val(DB* dbCxt, int key) {
+  return NULL;
+}
+
+
+/*
+ *  initializes database.
+ *  can only be called once for single database application.
+ *  @return - pointer to database context if success, NULL if failure.
+ */
+DB* db_init() {
+  DB* newDb = (DB*)malloc(sizeof(DB));
+  printf("dbManager initialized!!\n");
+  return newDb;
+}
+
+
+/*
+ *  destroys database.
+ *  @dataBase - context of the database to be operated on.
+ *  @return - 0 if success, -1 if failure.
+ */
+int db_destroy(DB* dataBase) {
+  if(!dataBase) return -1;
+  delete_tree(dataBase->head);
+  free(dataBase);
+  return 0;
+}
+
+/*  function to insert a key value pair into the database.
+ *  @dataBase - context of the database to be operated on.
  *  @key - a unique integer value.
  *  @val - pointer to the data stored.
  *  @return - 0 if success, -1 if failure.
  */
-dbNode* create_node(int key, void* val);
-DB* db_init() {
-  printf("dbManager initialized\n");
-  return NULL;
+int db_add_entry(DB* dbCxt, int key, void* val) {
```

Comment:
Ensure that the argument names are consistent with the comment and the header file.